### ISSUE TITLE
Text in Hero is cut

### ISSUE THREAD

**Issue by Kleak** [flutter/flutter#10246]
I have a hero transition between two pages and it seems there is a problem with the text.
Both text have the same TextStyle so it should not need to resize it just move it.

The text seems to be cut or the last letter go to a new line depending if softWrap if true or false.

here is the code: 
```dart
import 'dart:async';

import 'package:flutter/material.dart';

class Test extends StatefulWidget {
  @override
  TestState createState() => new TestState();
}

class TestState extends State<Test> {
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(routes: {
      "/": (BuildContext context) => new Login(),
      Home.path: (BuildContext context) => new Home(),
    });
  }
}

class Login extends StatefulWidget {
  @override
  _LoginState createState() => new _LoginState();
}

class _LoginState extends State<Login> {
  @override
  Widget build(BuildContext context) {
    final double width = MediaQuery.of(context).size.width;
    final container = new Container(
      decoration: new BoxDecoration(color: Colors.blue),
      padding: new EdgeInsets.symmetric(vertical: 100.0),
      width: width,
      child: new Column(
        children: [
          new Hero(
            tag: "hero-tag-title",
            child: new Text(
              "Test",
              overflow: TextOverflow.clip,
              softWrap: false,
              textAlign: TextAlign.center,
              style: new TextStyle(
                color: Colors.white,
                fontSize: 22.0,
                fontWeight: FontWeight.bold,
                decoration: TextDecoration.none,
              ),
            ),
          ),
          new RaisedButton(
            onPressed: () {
              Navigator.of(context).pushNamed("/home");
            },
            child: new Text("Login"),
          )
        ],
      ),
    );

    return new Scaffold(
      body: container,
    );
  }
}

class Home extends StatefulWidget {
  static const path = "/home";

  @override
  _HomeState createState() => new _HomeState();
}

class _HomeState extends State<Home> with TickerProviderStateMixin {
  List<MyCard> cards = [];

  void initState() {
    super.initState();
    new Future.delayed(new Duration(seconds: 2), () {
      _getCards();
    });
  }

  Future<Null> _getCards() async {
    for (int i = 1; i <= 10; i++) {
      MyCard card = new MyCard(
        i: i,
      );
      cards.add(card);
    }
    if (mounted) {
      setState(() {});
    }
  }

  @override
  Widget build(BuildContext context) {
    final appBar = new AppBar(
      centerTitle: true,
      primary: true,
      title: new Hero(
        tag: "hero-tag-title",
        child: new Text(
          "Test",
          textAlign: TextAlign.center,
          softWrap: false,
          style: new TextStyle(
            color: Colors.white,
            fontSize: 22.0,
            fontWeight: FontWeight.bold,
            decoration: TextDecoration.none,
          ),
        ),
      ),
    );
    if (cards.isEmpty) {
      return new Scaffold(
        appBar: appBar,
        body: new Center(
          child: new CircularProgressIndicator(),
        ),
      );
    }
    return new Scaffold(
      appBar: appBar,
      body: new ListView.builder(
        padding: new EdgeInsets.only(bottom: 10.0),
        itemCount: cards.length,
        itemBuilder: (BuildContext context, int index) {
          return cards[index];
        },
      ),
    );
  }
}

class MyCard extends StatelessWidget {
  final int i;

  MyCard({this.i});

  @override
  Widget build(BuildContext context) {
    return new Container(
      margin: new EdgeInsets.only(left: 15.0, top: 10.0, right: 15.0),
      child: new Card(
        child: new Container(
          height: 200.0,
          child: new Center(
            child: new Center(
              child: new Text("Card $i"),
            ),
          ),
        ),
      ),
    );
  }
}
```

and here a video :
![text-hero](https://cloud.githubusercontent.com/assets/5434752/26316529/d4ae1468-3f14-11e7-8e51-af7b37b84697.gif)

**Comment by m-jahanbakhsh** [comment]
any solution?
i have same problem

**Comment by duminos** [comment]
wrap it in a column?

**Comment by LongCatIsLooong** [comment]
Dup of #36220. This issue is older but since I assigned #36220 to myself I'm going to close this instead. This should be fixed once #36220 is closed.
